geopandas/tests/test_overlay.py

```diff
@@ -293,3 +293,14 @@ def test_preserve_crs(dfs, how):
     df2.crs = crs
     result = overlay(df1, df2, how=how)
     assert result.crs == crs
+
+def test_empty_intersection(dfs):
+    df1, df2 = dfs
+    polys3 = GeoSeries([Polygon([(-1,-1), (-3,-1), (-3,-3), (-1,-3)]),
+                        Polygon([(-3,-3), (-5,-3), (-5,-5), (-3,-5)])])
+    df3 = GeoDataFrame({'geometry': polys3, 'col3':[1,2]},
+                        crs={'init': 'epsg:4326', 'no_defs': True})
+    expected = GeoDataFrame([], columns=['col1', 'col3', 'geometry'])
+    result = overlay(df1, df3)
+    assert_geodataframe_equal(result, expected, check_column_type=False,
+                              check_less_precise=True, check_crs=False)
```

Comment:
`check_less_precise=True` should not be needed? (the same for `check_crs=False`)

Also, I find it a bit strange if `check_column_type=False` would also be needed to get this passing, as the code creates the empty frame in a very similar manner.

Reply:
The test kept failing, which is why I added the extra options....still failed. I was going to try the test first on my laptop, but for some reason the test is not finding ``pyproj``.

Reply:
I quickly tested it out, and for the test passes with only the `, check_like=True` option, which is needed because the column order is different for expected and result (which is probably something that can be solved as well, but didn't look into that yet)